### ISSUE TITLE
Point At Screen Emoji (specifically) doesn't show skin colours in skin colour popup

### ISSUE THREAD

**Issue by NamesNetworkNic** [florisboard/florisboard#1830]
This is a very visual issue so to save time I'm just going to show imagery.

Here's a screenshot showing the issue:
![Screenshot_2022-05-04-09-23-31-37_3aea4af51f236e4932235fdada7d1643](https://user-images.githubusercontent.com/68761174/166726709-43bb3bb9-8cde-4ff1-a03d-a909e7973f38.jpg)

Here's the next emoji over working as intended:
![Screenshot_2022-05-04-09-23-44-49_3aea4af51f236e4932235fdada7d1643](https://user-images.githubusercontent.com/68761174/166727050-d755d82a-7473-4b2b-97db-670aa765a857.jpg)

And here's a screen recording demonstrating the issue:

https://user-images.githubusercontent.com/68761174/166728725-e68d7d01-05d5-41ac-8708-5e50c8db63ee.mp4

This is on Florisboard "0.3.15 (80)" according to the app.
Android v12 on a Oneplus 9, latest security update as of posting.

I've never noticed this issue previously so it _could_ be new, but it's also an abscure problem and I may have have simply overlooked it in the past.

I hope my theme didn't make the problem too illegible, I noticed this while doing some design tweaks so it's a bit low contrast for the minute.

xo Nicolete

Edit: corrected the name of the emoji in the title from "Punch" to "Point At Screen" (they look similar, I'm sorry).

**Comment by patrickgold** [comment]
Interesting bug. This emoji which causes the issue seems to be quite a new one, neither my Android 11 OnePlus nor my PC can render it like at all.

Looking in the extracted CLDR data of the emojis, below block is the emoji and variants of it which causes the issue, right? (ignore the semicolons)

```
🫵;;
	🫵🏻;;
	🫵🏼;;
	🫵🏽;;
	🫵🏾;;
	🫵🏿;;
```

Should you not be able to see the emojis above correctly, could you please include the emoji in question in a GitHub comment so I can extract the correct code points for it? Thanks!

**Comment by NamesNetworkNic** [comment]
I can see the emojis you typed just fine (and yes those are the correct ones); they also appear correctly when I type them (see video) but they appear incorrectly in the menu.

Actually, I was about to type them to prove my point but it appears that they now show correctly in the menu - I wonder if this has something to do with how I was theming my keyboard at the time when I noticed the bug, or if it only appears when the default skin colour is yellow (it's the lightest shade now)?

I'm very confused now...


The emoji in question typed on my keyboard to rule out Android as the cause:

🫵
🫵🏻
🫵🏼
🫵🏽
🫵🏾
🫵🏿


New screenshot showcasing the further developement:

![IMG_20220505_122833](https://user-images.githubusercontent.com/68761174/167011160-c8795a5e-7c64-40fa-99d9-8df4f82a9d4e.jpg)


xo - Nicolete

**Comment by patrickgold** [comment]
Sorry for the late reply, got distracted with other issues.

So if I read your comment correctly, it now shows correctly, even if the base skin tone is yellow? If this normally does not occur, is it in any way reproducible?

Also an update from my side: apps that use emoji compat and request meta data version 8 or above cause my phone to also include that emoji, and on my end the emoji shows fine, regardless of which base skin tone I have.

**Comment by NamesNetworkNic** [comment]
Hi Patrick,
sorry for taking so long to reply, I've had a busy week.
I think you misunderstood what I wrote in my comment, although to be fair I wasn't very clear.

I haven't tested whether changing my default skin colour to yellow would bring the issue back, all I do know is that so far I've only seen it occur when the default skin tone is yellow, and it does not occur when the default skin colour is the lightest shade.

I would change the default back to yellow but quite frankly I don't know how.
If there is a way to change it I'll happily check for you if the update has had any effect and if I can reproduce the issue.

xo - Nicolete

**Comment by patrickgold** [comment]
> I haven't tested whether changing my default skin colour to yellow would bring the issue back, all I do know is that so far I've only seen it occur when the default skin tone is yellow, and it does not occur when the default skin colour is the lightest shade.

Hmm very strange. Maybe it was a one-time error, honestly don't know.

> I would change the default back to yellow but quite frankly I don't know how.

Floris Settings > Emoji > Default emoji skin tone. Here you can change the default skin tone.

**Comment by NamesNetworkNic** [comment]
Good news!

I've tested it now and the bug no longer seems present on the newest version (v0.3.15).

New screenshot:
![Screenshot_2022-06-09-01-18-50-64_affa6eecbd371edd54695182a950eba9](https://user-images.githubusercontent.com/68761174/172800318-b1054a94-9043-46b9-a428-72e4eb6c1be9.jpg)

Unless the bug returns I expect it has been fixed.
If it returns I'll reopen this issue and keep you updated.

Thank you very Patrick for your hard work, both on this bug and on Florisboard in general.

xo - Nicolete
